codegen-core/src/test/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/EnumGeneratorTest.kt

```diff
@@ -426,4 +427,133 @@ class EnumGeneratorTest {
         val variant3AsVariant3 = "SomeEnum::Variant3"
         expectMatchExpressionCompiles(modelV2, "test#SomeEnum", variant3AsVariant3)
     }
+
+    @Test
+    fun `impl debug for non-sensitive enum should implement the derived debug trait`() {
+        val model = """
+            namespace test
+            @enum([
+                { name: "Foo", value: "Foo" },
+                { name: "Bar", value: "Bar" },
+            ])
+            string SomeEnum
+        """.asSmithyModel()
+
+        val shape = model.lookup<StringShape>("test#SomeEnum")
+        val trait = shape.expectTrait<EnumTrait>()
+        val provider = testSymbolProvider(model)
+        val project = TestWorkspace.testProject(provider)
+        project.withModule(RustModule.Model) {
+            val generator = EnumGenerator(model, provider, this, shape, trait)
+            generator.render()
+            unitTest(
+                "impl_debug_for_non_sensitive_enum_should_implement_the_derived_debug_trait",
+                """
+                assert_eq!(format!("{:?}", SomeEnum::Foo), "Foo");
+                assert_eq!(format!("{:?}", SomeEnum::Bar), "Bar");
+                assert_eq!(
+                    format!("{:?}", SomeEnum::from("Baz")),
+                    "Unknown(UnknownVariantValue(\"Baz\"))");
```

Comment:
Try to adhere to rustfmt guidelines:

```suggestion
                assert_eq!(
                    format!("{:?}", SomeEnum::from("Baz")),
                    "Unknown(UnknownVariantValue(\"Baz\"))"
                );
```